# Change hard-coded /var paths to use /tmp instead
# @verdurin


diff -ur rstudio-1.4.1717/src/cpp/server/extras/conf/database.conf rstudio-1.4.1717.new/src/cpp/server/extras/conf/database.conf
--- rstudio-1.4.1717/src/cpp/server/extras/conf/database.conf	2021-05-24 22:22:25.000000000 +0000
+++ rstudio-1.4.1717.new/src/cpp/server/extras/conf/database.conf	2021-10-27 10:43:08.492652955 +0000
@@ -14,7 +14,7 @@
 #provider=sqlite
 #
 # Directory in which the sqlite database will be written
-#directory=/var/lib/rstudio-server
+#directory=/tmp/rstudio-server
 #
 # =================================================================================================================
 # postgresql configuration
diff -ur rstudio-1.4.1717/src/cpp/server/include/server/ServerOptions.gen.hpp rstudio-1.4.1717.new/src/cpp/server/include/server/ServerOptions.gen.hpp
--- rstudio-1.4.1717/src/cpp/server/include/server/ServerOptions.gen.hpp	2021-05-24 22:22:25.000000000 +0000
+++ rstudio-1.4.1717.new/src/cpp/server/include/server/ServerOptions.gen.hpp	2021-10-27 10:45:06.517626870 +0000
@@ -77,7 +77,7 @@
       value<bool>(&serverDaemonize_)->default_value(core::system::effectiveUserIsRoot()),
       "Indicates whether or not the rserver process should run as a daemon.")
       ("server-pid-file",
-      value<std::string>(&serverPidFile_)->default_value("/var/run/rstudio-server.pid"),
+      value<std::string>(&serverPidFile_)->default_value("/tmp/rstudio-server.pid"),
       "The path to a file where the rserver daemon's pid is written.")
       ("server-app-armor-enabled",
       value<bool>(&deprecatedServerAppArmorEnabled_)->default_value(false),
@@ -89,7 +89,7 @@
       value<std::string>(&secureCookieKeyFile_)->default_value(""),
       "If set, overrides the default path of the secure-cookie-key file used for encrypting cookies.")
       ("server-data-dir",
-      value<std::string>(&serverDataDir_)->default_value("/var/run/rstudio-server"),
+      value<std::string>(&serverDataDir_)->default_value("/tmp/rstudio-server"),
       "Path to the data directory where RStudio Server will write run-time state.")
       ("server-add-header",
       value<std::vector<std::string>>(&serverAddHeaders_)->default_value(std::vector<std::string>())->multitoken(),
diff -ur rstudio-1.4.1717/src/cpp/server/ServerOptions.cpp rstudio-1.4.1717.new/src/cpp/server/ServerOptions.cpp
--- rstudio-1.4.1717/src/cpp/server/ServerOptions.cpp	2021-05-24 22:22:25.000000000 +0000
+++ rstudio-1.4.1717.new/src/cpp/server/ServerOptions.cpp	2021-10-27 10:42:49.636816908 +0000
@@ -163,7 +163,7 @@
 
    // special program offline option (based on file existence at 
    // startup for easy bash script enable/disable of offline state)
-   serverOffline_ = FilePath("/var/lib/rstudio-server/offline").exists();
+   serverOffline_ = FilePath("/tmp/rstudio-server/offline").exists();
 
    // generate monitor shared secret
    monitorSharedSecret_ = core::system::generateUuid();
diff -ur rstudio-1.4.1717/src/cpp/server/server-options.json rstudio-1.4.1717.new/src/cpp/server/server-options.json
--- rstudio-1.4.1717/src/cpp/server/server-options.json	2021-05-24 22:22:25.000000000 +0000
+++ rstudio-1.4.1717.new/src/cpp/server/server-options.json	2021-10-27 10:45:36.128369528 +0000
@@ -54,7 +54,7 @@
             "name": "server-pid-file",
             "memberName": "serverPidFile_",
             "type": "string",
-            "defaultValue": "/var/run/rstudio-server.pid",
+            "defaultValue": "/tmp/rstudio-server.pid",
             "description": "The path to a file where the rserver daemon's pid is written."
          },
          {
@@ -85,7 +85,7 @@
             "name": "server-data-dir",
             "memberName": "serverDataDir_",
             "type": "core::FilePath",
-            "defaultValue": "/var/run/rstudio-server",
+            "defaultValue": "/tmp/rstudio-server",
             "description": "Path to the data directory where RStudio Server will write run-time state."
          },
          {
diff -ur rstudio-1.4.1717/src/cpp/server_core/ServerDatabase.cpp rstudio-1.4.1717.new/src/cpp/server_core/ServerDatabase.cpp
--- rstudio-1.4.1717/src/cpp/server_core/ServerDatabase.cpp	2021-05-24 22:22:25.000000000 +0000
+++ rstudio-1.4.1717.new/src/cpp/server_core/ServerDatabase.cpp	2021-10-27 10:40:59.800771880 +0000
@@ -44,7 +44,7 @@
 constexpr const char* kDatabaseProviderSqlite = "sqlite";
 constexpr const char* kDatabaseProviderPostgresql = "postgresql";
 constexpr const char* kSqliteDatabaseDirectory = "directory";
-constexpr const char* kDefaultSqliteDatabaseDirectory = "/var/lib/rstudio-server";
+constexpr const char* kDefaultSqliteDatabaseDirectory = "/tmp/rstudio-server";
 constexpr const char* kDatabaseHost = "host";
 constexpr const char* kDefaultDatabaseHost = "localhost";
 constexpr const char* kDatabaseName = "database";
